### ISSUE TITLE
Can't use brand-new Scala release

### ISSUE THREAD

**Issue by SethTisue** [VirtusLab/scala-cli#1090]
```
% scala-cli about
Scala CLI version 0.1.7
% scala-cli repl -S 2.12.16
[error]  Cannot find matching Scala version for '2.12.16'
You can only choose one of the 3.x, 2.13.x, and 2.12.x. versions.
The latest supported stable versions are 3.1.2, 2.13.8, 2.12.15.
In addition, you can request compilation with the last nightly versions of Scala,
by passing the 2.nightly, 2.12.nightly, 2.13.nightly, or 3.nightly arguments.
Specific Scala 2 or Scala 3 nightly versions are also accepted.
```

this is even though 2.12.16 is available on Maven Central

I guess that scala-cli has some kind of cache of what Scala versions exist? if so, how do I force that cache to be rebuilt?

and, maybe a cache miss should _always_ force the cache to rebuild?

**Comment by alexarchambault** [comment]
It might originate from stale file listings in cache. Can you try with `--ttl 0s`?

`scala-cli repl -S 2.12.16` or things like `echo 'println("Hello")' | scala-cli -S 2.12.16 -` work for me locally.

**Comment by SethTisue** [comment]
`--ttl 0s` does the trick, yeah, thanks. but I still wonder:

> maybe a cache miss should always force a retry

**Comment by lwronski** [comment]
Thanks for reporting, it should be fixed by #1096
